### DISCUSSION TITLE
Get search results from a specific list

### DISCUSSION THREAD

**Discussion by claudivieira** [microsoft-search/pnp-modern-search#1397] [Q&A] [answered]
**Version used**
Ex: 4.2

**Describe the bug**
Hello everyone. I'm trying to have a search results page based on a list. 
I have a list of names and I need to be able to search in the box for whatever name in the list. I've tried several things that I saw over the internet, but I can't make it work only with results from that specific list so I thought maybe you could give me some help over here or tell me what I'm missing. TIA

**Screenshots**
![image](https://user-images.githubusercontent.com/39301758/137173147-53710e01-4091-46bf-9ad9-a91b9835c70c.png)

**Desktop (please complete the following information):**
 - Browser: Chrome and Firefox

**Comment by wobba** [comment, marked as answer]
Change the query template to: `path:https://contoso.sharepoint.com/Lists/Clients contentclass:sts_listitem_* ` this will return only list items from your list

**Reply by claudivieira** [reply]
Hi @wobba , thanks a lot for the quick reply. I've tried it and it doesn't work. Could it be that I'm missing other settings?

**Reply by FranckyC** [reply]
What about `{searchTerms}* Path:"https://contoso.sharepoint.com/Lists/Clients" contentclass:sts_listitem_*`?

**Reply by claudivieira** [reply]
Also no :( @FranckyC 
I'm going to share some sreenshots of the settings for the search box and search results I have, maybe it helps to identify something wrong

![image](https://user-images.githubusercontent.com/39301758/137185052-93263fec-88c1-47ff-ac67-0f664cde9de3.png)
![image](https://user-images.githubusercontent.com/39301758/137185114-de1d144c-6ec6-464c-8d69-7c5e00893930.png)
![image](https://user-images.githubusercontent.com/39301758/137185307-49bb03cd-1d9d-4ed3-bde0-c14b3c188f85.png)
![image](https://user-images.githubusercontent.com/39301758/137185399-9ada444b-a04b-4d21-9201-e19804c3cd71.png)
![image](https://user-images.githubusercontent.com/39301758/137185426-cc973f3b-3b2b-4e8b-a70d-3bdac10ca719.png)
![image](https://user-images.githubusercontent.com/39301758/137185476-05c565c9-0e3b-416b-a216-5786fbb2e0dc.png)

**Reply by FranckyC** [reply]
Ok I see a wrong configuration at search box level. Please turn off the "Use dynamic data source as default input" and retry.

**Reply by claudivieira** [reply]
nothing, it keeps saying there are no results

**Reply by FranckyC** [reply]
Ok then let's go step by step. First add a Search Results on the page with the following query `Path:"https://contoso.sharepoint.com/Lists/Clients" contentclass:sts_listitem_*`? Do you see any results?

**Reply by FranckyC** [reply]
Also I assume you replaced the `https://contoso.sharepoint.com` by your real SharePoint URL ;). Otherwise it can't really work (unless you actually work for Contoso).

**Reply by claudivieira** [reply]
Yes of course :D I have my real URL there. I deleted the webpart and added again, added the query on top and it still says no results
![image](https://user-images.githubusercontent.com/39301758/137192661-413201ec-4554-4af2-b272-615417e898e0.png)

**Reply by FranckyC** [reply]
Weird. Can you try the same with the default "Highlighted Content" Web Part to see if there is actually results in your list? The query seems valid.

**Reply by claudivieira** [reply]
Sure, should I add the same query on the Custom Query option of the webpart?

**Reply by claudivieira** [reply]
Looks like now it works, but seems that I was using the wrong list link (I mean the URL was right but not for the Site I needed) I'm so sorry! 😱 
It's solved now! Thanks a lot @FranckyC

**Reply by FranckyC** [reply]
Haha no worries ;). Glad it is solved now.

**Reply by claudivieira** [reply]
Hi again :)
I was wondering if it would be possible be have access to any of the properties inside the list I'm using to query :) 
I've checked the API response and the properties being returned are not the ones from the list (neither the main one, which is being returned as Title instead as Name). For example, one of the List Columns is called 'Address', and I would like to show it when showing the search results query. Would this be possible somehow?

Thanks once again

**Reply by sayuri8** [reply]
I also tried this it works but I need only a specific list and not all available list. Now it show also items from other lists. What I am doing wrong?

**Reply by wobba** [reply]
@sayuri8 the above with `path` will limit to one particular list. Please ask a separate question if this is not working for you, as it's likely something configured wrong.

**Reply by sayuri8** [reply]
I see then I will create a new thread.

**Reply by dnenov** [reply]
What a self-absorbed answer. A cryptic one-liner. Why is it contentclass:sts_listitem_* , where does it say, any links to a step-by-step guide on how to create a rudimentary functionality that should be there to begin with??? No. I am a big-shot developer with a fat salary, and the end-user is my enemy. Well done.

**Reply by wobba** [reply]
@dnenov Thank you for your insightful comment. Content class is a 20 year old SharePoint concept, not very well documented. But here is a starting point for you. https://learn.microsoft.com/en-us/previous-versions/office/developer/sharepoint2001/dd588220(v=office.11)

No one will disagree with you that something should be simple. However these web parts are a replacement for the classic search web parts, which were equally hard to set up, and not really meant for end users. So any one liner shared is accumulated knowledge. Sure I could write a long paragraph about all there is about contentclass, but that would be somewhat out of scope as well. If you search around you will find numerous blog posts on contentclass values, since the official Microsoft documentation is lacking.

Best of luck on creating awesome solutions for end users.

**Reply by dnenov** [reply]
My apologies, I should not have spoken in frustration and offended you in your attempt to help the community and bring knowledge. I apologize.

**Reply by wobba** [reply]
@dnenov I don't offend easily, and it's a fair question to ask how contentclass work and how my answer solves the question.

Every list item will start with the prefix `sts_listitem_` and every list with `sts_list_`. If you request contentclass as a refiner/filter you can learn what values exists, and also examine the value of this property for different items. Like a site has `sts_site` and a sub-site has `sts_web` to list some common ones. And again, web search is your friend here as official documentation is rather sparse in the area.
